feat(vite-plugin-angular): add support for import attributes

joshuamorony opened the pull request:
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/analogjs/analog/blob/main/CONTRIBUTING.md#-commit-message-guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Which package are you modifying?

- [x] vite-plugin-angular
- [ ] vite-plugin-nitro
- [ ] astro-angular
- [ ] create-analog
- [ ] router
- [ ] platform
- [ ] content
- [ ] nx-plugin
- [ ] trpc

## What is the current behavior?

See: https://github.com/analogjs/analog/discussions/901

## What is the new behavior?

Allows mapping `providers`, `viewProviders`, `imports`, and `exposes` to component/directive metadata via import attributes (`exposes` is a special case that has special handling and is not added to the metadata)

## Does this PR introduce a breaking change?

- [x] Yes
- [ ] No

This removes the existing functionality that supported adding analog components to `imports` if the import was directly a `.analog` file. We could leave this functionality in place also if we want.

## Other information

## [optional] What [gif](https://chrome.google.com/webstore/detail/gifs-for-github/dkgjnpbipbdaoaadbdhpiokaemhlphep) best describes this PR or how it makes you feel?

<img src="https://media3.giphy.com/media/lNRIshllxjpSy2FpVV/giphy.gif"/>

nartc commented on packages/vite-plugin-angular/src/lib/authoring/analog.ts:
```diff
@@ -160,10 +167,36 @@ function processAnalogScript(
+            );
+          }
+
+          if (attribute.name === 'meta') {
```
nit: we should probably check this along with the above condition?

nartc commented on packages/vite-plugin-angular/src/lib/authoring/analog.ts:
```diff
@@ -160,10 +167,36 @@ function processAnalogScript(
+          }
+
+          if (attribute.name === 'meta') {
+            const defaultImport = node.getDefaultImport();
```
nit: the `getStructure()` above should have the default import in there.

nartc commented on packages/vite-plugin-angular/src/lib/authoring/analog.ts:
```diff
@@ -160,10 +167,36 @@ function processAnalogScript(
+              meta[value].push(defaultImport.getText());
+            }
+
+            const namedImports = node.getNamedImports();
```
nit: same as above.the `getStructure()` should have this information already

nartc reviewed (commented):
A couple of nits and a question/suggestion

Long shot but what if an `ImportAttribute` is used purposefully from the userland that don't have anything to do with our `meta` checking? We probably should still keep those `ImportAttribute` around for their import statements when we move those imports to a `.ts` file.

joshuamorony commented:
> Long shot but what if an `ImportAttribute` is used purposefully from the userland that don't have anything to do with our `meta` checking? We probably should still keep those `ImportAttribute` around for their import statements when we move those imports to a `.ts` file.

Yes this makes sense - I've removed the error what was thrown if it doesn't match what Analog expects so non-Analog import attributes can still pass through. I replaced the error with a `console.warn`, is there a better approach here?
